lightning/src/util/events.rs

```diff
@@ -81,6 +82,39 @@ impl_writeable_tlv_based_enum!(PaymentPurpose,
 	(2, SpontaneousPayment)
 );
 
+/// When the payment failure took place and extra details about it. [`Failure::OnPath`] may contain
+/// a [`NetworkUpdate`] that needs to be applied to the [`NetworkGraph`].
+///
+/// [`NetworkUpdate`]: crate::routing::gossip::NetworkUpdate
+/// [`NetworkGraph`]: crate::routing::gossip::NetworkGraph
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Failure {
```

Comment:
I think we need a more specific name here if this is in a general `events` module. Maybe `HtlcFailure`?

Reply:
I'll go with `PathFailure` since there's already `onion_utils::HTLCFailureReason`